chainer/functions/math/batch_l2_norm_squared.py

```diff
@@ -35,12 +35,20 @@ def forward_gpu(self, inputs):
         )
         return l2normsquared_kernel(x, axis=1),
 
-    def backward(self, inputs, gy):
+    def backward_cpu(self, inputs, gy):
         x = _as_two_dim(inputs[0])
         xp = cuda.get_array_module(x, gy)
         return xp.array([2 * x[i] * gy[0][i] for i in six.moves.range(len(x))],
                         dtype=np.float32).reshape(inputs[0].shape),
 
+    def backward_gpu(self, inputs, gy):
+        x = _as_two_dim(inputs[0])
+        gy_repeat = gy[0].reshape(len(gy[0]), 1).repeat(x.shape[1], axis=1)
```

Comment:
I'm not really happy with repeating the grad array like this, but it was the only way I found to feed it into the elementwise kernel. Can you suggest a better way?